component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -443,16 +461,59 @@ private boolean isBackupCodesExistForUser(String username)
     }
 
     private String buildBackupCodeLoginPageURL(AuthenticationContext context, String username, String retryParam,
-                                               String multiOptionURI)
+                                               String errorParam, String multiOptionURI)
             throws AuthenticationFailedException, URISyntaxException, URLBuilderException {
 
         String queryString = "sessionDataKey=" + context.getContextIdentifier() + "&authenticators=" + getName() +
-                "&type=backup-code" + retryParam + "&username=" + username + multiOptionURI;
+                "&type=backup-code" + retryParam + "&username=" + username + errorParam + multiOptionURI;
         String loginPage = FrameworkUtils.appendQueryParamsStringToUrl(BackupCodeUtil.getBackupCodeLoginPage(context),
                 queryString);
         return buildAbsoluteURL(loginPage);
     }
 
+    private String buildErrorParamString(Map<String, String> paramMap) {
+
+        StringBuilder params = new StringBuilder();
+        for (Map.Entry<String, String> entry : paramMap.entrySet()) {
+            params.append("&").append(entry.getKey()).append("=").append(entry.getValue());
+        }
+        return params.toString();
+    }
+
+    private String getErrorParamsStringFromErrorContext() {
+
+        String errorParam = "";
+        IdentityErrorMsgContext errorContext = IdentityUtil.getIdentityErrorMsg();
+        IdentityUtil.clearIdentityErrorMsg();
+        if (errorContext != null && errorContext.getErrorCode() != null) {
+
+            log.debug("Identity error message context is not null");
+
+            String errorCode = errorContext.getErrorCode();
+            String reason = null;
+
+            String[] errorCodeWithReason = errorCode.split(":", 2);
+            errorCode = errorCodeWithReason[0];
+            if (errorCodeWithReason.length > 1) {
+                reason = errorCodeWithReason[1];
+
```

Comment:
unnecessary new line

Reply:
Addressed in 807d553798051918aaacd7d3f11357bb3eceec73